pyproject.toml

```diff
@@ -0,0 +1,67 @@
+[tool.ruff]
+# Exclude a variety of commonly ignored directories.
+exclude = [
+    ".bzr",
+    ".direnv",
+    ".eggs",
+    ".git",
+    ".git-rewrite",
+    ".hg",
+    ".ipynb_checkpoints",
+    ".mypy_cache",
+    ".nox",
+    ".pants.d",
+    ".pyenv",
+    ".pytest_cache",
+    ".pytype",
+    ".ruff_cache",
+    ".svn",
+    ".tox",
+    ".venv",
+    ".vscode",
+    "__pypackages__",
+    "_build",
+    "buck-out",
+    "build",
+    "dist",
+    "docs/src",
+    "node_modules",
+    "site-packages",
+    "venv",
+]
+
+# Same as Black.
+line-length = 120
+indent-width = 4
+
+# Assume Python 3.8
+target-version = "py38"
```

Comment:
Would this cause any trouble? I think the version depends on user's system.

Reply:
According to https://github.com/astral-sh/ruff/issues/2519#issuecomment-1467101887, is ok to indicate the minimum version, which according to `setup.py` is `3.9`, so I added it in https://github.com/pyro-ppl/numpyro/pull/1700/commits/b2292eef29fd517219f3d15a78aa69524455a68c

Reply:
In https://github.com/astral-sh/ruff/blob/main/CHANGELOG.md#015 it's recommended to use `project.requires-python`

Reply:
Changed in https://github.com/pyro-ppl/numpyro/pull/1700/commits/442353e2c26659c0d062728385f26721a8210508